dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerConnector.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+
+@AutoService({Connector.class})
+@Description("Dumps metadata from Cloudera Manager.")
+@RespectsArgumentUser
+@RespectsArgumentPassword
+@RespectsArgumentUri
+public class ClouderaManagerConnector extends AbstractConnector {
+
+  public static final String FORMAT_NAME = "cloudera-manager.dump.zip";
+
+  public ClouderaManagerConnector() {
+    super("cloudera-manager");
+  }
+
+  @Nonnull
+  @Override
+  public String getDefaultFileName(boolean isAssessment, Clock clock) {
+    return ArchiveNameUtil.getFileNameWithTimestamp(getName(), clock);
+  }
+
+  @Override
+  public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArguments arguments)
+      throws Exception {
+    out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
+    out.add(new FormatTask(FORMAT_NAME));
+    out.add(new ClouderaClustersTask());
+    out.add(new ClouderaCMFHostsTask());
+    out.add(new ClouderaAPIHostsTask());
+    out.add(new ClouderaServicesTask());
+  }
+
+  @Nonnull
+  @Override
+  public ClouderaManagerHandle open(@Nonnull ConnectorArguments arguments) throws Exception {
+    List<String> errors = new ArrayList<>();
+    if (arguments.getUri()
+        == null) { // todo --url currently used for JDBC drivers, think about another param
+      errors.add("--url for Cloudera Manager API is required");
+    }
+    if (arguments.getUser() == null) {
+      errors.add("--user is required for Cloudera Manager API connector");
+    }
+
+    if (!errors.isEmpty()) {
+      throw new MetadataDumperUsageException(
+          "Missing arguments for connector generic-args :", errors);
+    }
+    URI uri = new URI(arguments.getUri());
+
+    CloseableHttpClient httpClient = disableSSLVerification(HttpClients.custom()).build();
+    ClouderaManagerHandle handle = new ClouderaManagerHandle(uri, httpClient);
+
+    String user = arguments.getUser();
+    String password = arguments.getPasswordOrPrompt();
+    doClouderaManagerLogin(handle.getBaseURI(), httpClient, user, password);
+
+    return handle;
+  }
+
+  private void doClouderaManagerLogin(
+      URI baseURI, CloseableHttpClient httpClient, String user, String password) throws Exception {
+    ClouderaManagerLoginHelper loginHelper = new ClouderaManagerLoginHelper();
+    loginHelper.login(baseURI, httpClient, user, password);
+  }
+
+  private HttpClientBuilder disableSSLVerification(HttpClientBuilder builder) throws Exception {
+    // Cloudera Manager API SSL certificate is not in list of know certificates.
+    // So, switch off SSL certificate validation.
+    // It is  expected that Dumper will work  in internal private network (probably localhost calls)
+    builder.setSSLContext(
```

Comment:
Can't it be solved in an environment by adding the certificate to the trusted ones? Maybe it's worth to add a parameter that controls this behavior (in a separate pr)?

Reply:
Yes, it's possible. But if we will take into account our use case it will mean that our customers will need to do an additional not trivial step to use our tool. At the same time I don't see a lot of value on this because the tool should be used in their private datacenter/network. 

Let's keep it as is for usability simplification.